controllers/toolchaincluster/toolchaincluster_controller.go

```diff
@@ -61,5 +62,32 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return reconcile.Result{}, err
 	}
 
+	// add toolchaincluster role label if not present
+	reqLogger.Info("adding cluster role label based on type")
+	if err := r.addToolchainClusterRoleLabelFromType(reqLogger, toolchainCluster); err != nil {
+		return reconcile.Result{}, err
+	}
+
 	return reconcile.Result{}, r.clusterCacheService.AddOrUpdateToolchainCluster(toolchainCluster)
 }
+
+func (r *Reconciler) addToolchainClusterRoleLabelFromType(log logr.Logger, toolchainCluster *toolchainv1alpha1.ToolchainCluster) error {
+	if _, clusterTypeFound := toolchainCluster.Labels[cluster.LabelType]; !clusterTypeFound {
+		log.Info("cluster `type` label not found, unable to add toolchain cluster role label from type")
+		return nil
+	}
+	if toolchainCluster.Labels[cluster.LabelType] != string(cluster.Member) {
+		log.Info("cluster `type` is not member, skipping cluster role label setting")
+		return nil
+	}
```

Comment:
```suggestion
        if clusterType, found := toolchainCluster.Labels[cluster.LabelType]; !found {
		log.Info("cluster `type` label not found, unable to add toolchain cluster role label from type")
		return nil
	} else if clusterType != string(cluster.Member) {
		log.Info("cluster `type` is not member, skipping cluster role label setting")
		return nil
	}
```

Reply:
I'm not a big fan of `else` statement usage, to me it involves more thinking 😄 , but it works the same ! I can change it!

Reply:
changed in [5fd587a46e83cad1](https://github.com/mfrancisc/toolchain-common/commit/5fd587a46e83cad164b50f71c52895b9fce79610)

Reply:
thanks!